DOCS Add blurbs about several new features to 4.11 changelog (#10225)
silverstripe/silverstripe-framework: docs/en/04_Changelogs/4.11.0.md

## Features and enhancements {#features-and-enhancements}

### Preview any DataObject in any admin section {#cms-preview}

> The CMS preview panel has historically on been available on `Versioned` `DataObjects` in the Pages admin section. This has now been expanded to allow any `DataObject`(regardless of whethwr they jse the`Versioned` extension) to be previewed in any section of the CMS.

Author: ```suggestion
The CMS preview panel has historically on been available for `Versioned` `DataObjects` in the Pages admin section. This has now been expanded to allow any `DataObject`(regardless of whether they use the`Versioned` extension) to be previewed in any section of the CMS.
```

Reviewer: You know it's your PR so you can update these straight away, right? :-D

Author: I made this PR and found those on my tablet. I don't wanna make a bunch of little "fixed typo" commits and I don't have a way of doing a `git commit -amend` and `git force --force` without turning on my computer but my couch is so comfy :p

> The CMS preview panel has historically on been available for `Versioned` `DataObjects` in the Pages admin section. This has now been expanded to allow any `DataObject`(regardless of whether they use the`Versioned` extension) to be previewed in any section of the CMS.

Reviewer: ```suggestion
The CMS preview panel has historically been available for `Versioned` `DataObjects` in the Pages admin section. This has now been expanded to allow any `DataObject` (regardless of whether it uses the `Versioned` extension) to be previewed in any section of the CMS.
```

> This can be used to allow content authors to see the content they are creating in the context it will be presented to users. Example use cases include previewing `DataObjects` which belong to a page (e.g. the [dnadesign/silverstripe-elemental module](https://github.com/silverstripe/silverstripe-elemental) allows previewing elemental blocks which are not inline-editable), and previewing `DataObjects` in a custom admin section, such as email or system-generated PDF templates.

Author: ```suggestion
This can be used to allow content authors to see the content they are creating in the context it will be presented to users. Example use cases include previewing `DataObjects` which belong to a page (e.g. the [dnadesign/silverstripe-elemental module](https://github.com/silverstripe/silverstripe-elemental) allows previewing elemental blocks which are not inline-editable), and previewing `DataObjects` in a custom admin section, such as templates for emails or system-generated PDF documents.
```

> This can be used to allow content authors to see the content they are creating in the context it will be presented to users. Example use cases include previewing `DataObjects` which belong to a page (e.g. the [dnadesign/silverstripe-elemental module](https://github.com/silverstripe/silverstripe-elemental) allows previewing elemental blocks which are not inline-editable), and previewing `DataObjects` in a custom admin section, such as templates for emails or system-generated PDF documents.

Reviewer: `DataObjects` is not accurate here, though not sure whether to go with `DataObject` representing plural as well or do something like `DataObject`s.

Reviewer: Below you used `GridFieldComponent`s so we can go with that everywhere.

Author: Ahh yup I would have meant to use `DataObject`s originally. I'll fix that up

The [Preview Documentation](https://docs.silverstripe.org/en/4/developer_guides/customising_the_admin_interface/preview/) has been updated with code examples which show how to enable CMS preview on `DataObjects` in a couple of different scenarios.

### Other new features {#other-features}

> - A new [AbstractGridFieldComponent](https://api.silverstripe.org/4/SilverStripe/Forms/GridField/AbstractGridFieldComponent.html) class has been added to make it easier to globally add fundamental functionality to `GridFieldComponent`s. All classes packed with the Silverstripe framework which implement the `GridFieldComponent` interface are subclasses of the new abstract class, making them all `Injectable`. Maintainers of third-party packages which include classes that implement `GridFieldComponent` are encouraged to subclass the `AbstractGridFieldComponent` abstract class.

Author: ```suggestion
- A new [AbstractGridFieldComponent](https://api.silverstripe.org/4/SilverStripe/Forms/GridField/AbstractGridFieldComponent.html) class has been added to make it easier to globally add fundamental functionality to `GridFieldComponent`s. All classes packaged with the Silverstripe framework which implement the `GridFieldComponent` interface are subclasses of the new abstract class, making them all `Injectable`. Maintainers of third-party packages which include classes that implement `GridFieldComponent` are encouraged to subclass the `AbstractGridFieldComponent` abstract class.
```

> - New options have neen added to the [dnadesign/silverstripe-elemental module](https://github.com/silverstripe/silverstripe-elemental) to control what content is indexed for searching elemental blocks. see [the documentation](https://github.com/silverstripe/silverstripe-elemental/blob/4/docs/en/searching-blocks.md) for details.

Author: ```suggestion
- New options have been added to the [dnadesign/silverstripe-elemental module](https://github.com/silverstripe/silverstripe-elemental) to control what content is indexed for searching elemental blocks. see [the documentation](https://github.com/silverstripe/silverstripe-elemental/blob/4/docs/en/searching-blocks.md) for details.
```

> - New options have been added to the [dnadesign/silverstripe-elemental module](https://github.com/silverstripe/silverstripe-elemental) to control what content is indexed for searching elemental blocks. see [the documentation](https://github.com/silverstripe/silverstripe-elemental/blob/4/docs/en/searching-blocks.md) for details.

Reviewer: This might be helpful but probably shouldn't be a part of the changelog, what do you reckon @maxime-rainville ?

Author: This is based off the [4.8.0](https://docs.silverstripe.org/en/4/changelogs/4.8.0/) and [4.9.0](https://docs.silverstripe.org/en/4/changelogs/4.9.0/) changelogs which both had something like this.

Author: The reasons I added these specific features:
- Pointing out the new `AbstractGridFieldComponent` abstract class in changelogs gives a higher chance of it being adopted in third-party modules.
- IMO the new elemental block index features represent a reasonably substantial increase in control of indexed content, especially for sites for which page content is exclusively represented by elemental blocks.

Reviewer: Fair point. Maybe the heading just needs to change slightly (or split into two separate ones) to indicate it's a notable change in another module other than the framework or the CMS.

Author: Fair. Maybe something like "Other new features in [Supported Modules](https://www.silverstripe.org/software/addons/silverstripe-commercially-supported-module-list/)"?

Reviewer: That would work for me. Let's see what Max says.

Reviewer 2: I'd say add it in for now. 

Once we get closer to the release, we usually spend a bit more time reviewing the fine print to make sure everything is clear and concise. Probably not much point obsessing about it for now.

Author: So should I leave the heading as-is or change it?

Reviewer: Leave for now, thanks
